sympy/vector/functions.py

```diff
@@ -6,6 +6,101 @@
+        The coordinate system the expr is to be expressed in
+
+    Examples
+    =========
+
+    >>> from sympy.vector import extended_express, CoordSys3D
+    >>> from sympy import pi
+    >>> N=CoordSys3D('N')
+    >>> C=N.create_new('C',transformation='cylindrical',vector_names=('r','theta','z'))
+    >>> extended_express(5*N.i,C)
+    5*C.r
+    >>> extended_express(5*N.j,C)
+    5*C.r + pi/2*C.theta
+    >>> extended_express(10*N.k,C)
+    10*C.z
+    >>> extended_express(6*C.r,N)
+    6*N.i
+    >>> extended_express(6*C.r+pi/2*C.theta,N)
+    6*N.j
+
+    where a coordinate is undefined (eg theta with extended_express(N.k), it is mapped to zero
+    The non-cartesian frame must be defined as a child of the cartesian frame as above.
+
+    the output is expressed in terms of the base vectors of the derived coordinate systems
+    currently they default to i,j and k which are confusing for non-cartesian systems so it
+    is recommended to explicitly name them as above.
+
+    See Also
+    ========
+
+    CoordSys3D.transformation_from_parent,CoordSys3D.transformation_to_parent
+
+    """
+    if not isinstance(coordsys, CoordSys3D):
+        raise TypeError("system should be a CoordSys3D instance") # check we are expressing in a frame
+
+    parts=expr.separate() # get different coordinate frames in vector, cry if more than one
+
+    if isinstance(parts,dict):
+        if(len(parts)==0): return Vector.zero # we have been given zero vector
+        if(len(parts)>1):  raise ValueError("Does not support expressions containing multiple base coordinate frames")
+
+        foundframe=tuple(parts.keys())[0]
+        foundvector=tuple(parts.values())[0]
+    else:
+        foundframe=parts.system
+        foundvector=parts
+
+    # we should now have found the only frame in the vector, and we now have to convert it to the given frame
+
+    from_frame=foundframe
+    to_frame=coordsys
+
+    from_coeffs1=from_frame.base_vectors()
+    from_coeffs2=from_frame.base_scalars()
+    to_coeffs=to_frame.base_vectors()    # output with vectors
+
+    if from_frame._parent==to_frame:
+        transform_function=from_frame._transformation_lambda
+    else:
+        if to_frame._parent==from_frame:
+            transform_function=to_frame._transformation_from_parent_lambda
+        else:
+            raise ValueError("Cannot link Coordinate frames")
+    args=[]
+
+    for i,j in zip(from_coeffs1,from_coeffs2): # could be expressed in either for inertial frame
+        coeff1=foundvector.coeff(i)  # understand both N.i and N.x (or C.r and C.i)
+        coeff2=foundvector.coeff(j)
+        if coeff1!=0 and coeff2!=0: raise ValueError("Cannot express vector with both base vectors and base scalars - check your vector")
+        args.append(coeff1+coeff2) # only one can be non-zero
+
+    vals=transform_function(*args)
+
+    ans=Vector.zero
+
+    for v,c in zip(vals,to_coeffs):
+        if isnan(v):  # v is nan ie infinity from an arctan
```

Comment:
Transformation functions in SymPy should not return Python's floating point NaN. SymPy has its own `NaN` that is tested by `v is NaN`.

Reply:
agreed. couldn't find isnan for sympy, but there is a sympy.nan symbol that works